packages/channels/src/messenger/api.ts

```diff
@@ -0,0 +1,140 @@
+import crypto from 'crypto'
+import express, { Response, Request, NextFunction } from 'express'
+import { IncomingMessage } from 'http'
+import yn from 'yn'
+import { ChannelApi, ChannelApiManager, ChannelApiRequest } from '../base/api'
+import { ChannelInitializeEvent, ChannelStartEvent } from '../base/service'
+import { MessengerService } from './service'
+
+export class MessengerApi extends ChannelApi<MessengerService> {
+  // For legacy
+  private pageIdToScope: { [pageId: string]: string } = {}
+
+  async setup(router: ChannelApiManager) {
+    router.use('/messenger', express.json({ verify: this.prepareAuth.bind(this) }))
+    // Legacy stuff
+    if (yn(process.env.SPINNED)) {
+      router.use('/messenger', this.mapScope.bind(this))
+    }
+    router.get('/messenger', this.handleWebhookVerification.bind(this))
+
+    router.post('/messenger', this.auth.bind(this))
+    router.post('/messenger', this.handleMessageRequest.bind(this))
+
+    this.service.on('start', this.handleStart.bind(this))
+    this.service.on('initialize', this.handleInitialize.bind(this))
+  }
+
+  protected async handleStart({ scope }: ChannelStartEvent) {
+    // Legacy stuff
+    if (yn(process.env.SPINNED)) {
+      const { client } = this.service.get(scope)
+
+      try {
+        const pageId = await client.getPageId()
+        this.pageIdToScope[pageId] = scope
+      } catch {
+        // when in live mode this call can fail. we can work around it for new users since they are supposed to use the botId in the url
+        // we don't show an error because this is correct usage
+      }
+    }
```

Comment:
Legacy garbage. Migth stay here forever actually and we just don't bring it over to the next version of the channel